cmd/checkpoint/main.go

```diff
@@ -60,21 +60,64 @@ func main() {
 	}
 
 	glog.Infof("Starting checkpointer for node: %s", nodeName)
-	run(newClient(), nodeName)
+	// This is run as a static pod, so we can't use InClusterConfig (no secrets/service account)
+	// Use the same kubeconfig as the kubelet for auth and api-server location.
+	kubeConfig, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		&clientcmd.ClientConfigLoadingRules{ExplicitPath: kubeconfigPath},
+		&clientcmd.ConfigOverrides{}).ClientConfig()
+	if err != nil {
+		glog.Fatalf("Failed to load kubeconfig: %v", err)
+	}
+	client := clientset.NewForConfigOrDie(kubeConfig)
+
+	load := func(rawData []byte, filepath string) []byte {
+		if len(rawData) > 0 {
+			return rawData
+		}
+		data, err := ioutil.ReadFile(filepath)
```

Comment:
This is extracting it straight from the kubeconfig. I didn't want to assume that the kubeconfig inlined the certs.